packages/pluggableWidgets/accordion-native/src/components/AccordionGroup.tsx

```diff
@@ -0,0 +1,62 @@
+import { createElement, ReactElement } from "react";
+import { View, Pressable, Text } from "react-native";
+import { DynamicValue, NativeIcon } from "mendix";
+
+import { GroupIcon } from "./GroupIcon";
+import { AnimatedCollapsibleView } from "./CollapsibleView";
+import { GroupsType, IconEnum } from "../../typings/AccordionProps";
+import { AccordionGroupStyle } from "../ui/Styles";
+
+export interface AccordionGroupProps {
+    index: number;
+    collapsible: boolean;
+    icon: IconEnum;
+    iconCollapsed: DynamicValue<NativeIcon> | undefined;
+    iconExpanded: DynamicValue<NativeIcon> | undefined;
+    group: GroupsType;
+    isExpanded: boolean;
+    onPressGroupHeader: (group: GroupsType, index: number) => void;
+    visible: boolean;
+    style: AccordionGroupStyle;
+}
+
+export function AccordionGroup({
+    index,
+    collapsible,
+    icon,
+    iconCollapsed,
+    iconExpanded,
+    group,
+    isExpanded,
+    onPressGroupHeader,
+    visible,
+    style
+}: AccordionGroupProps): ReactElement | null {
+    return visible ? (
+        <View style={style.container}>
+            <Pressable
+                style={[style.header.container, icon === "left" && { flexDirection: "row-reverse" }]}
+                onPress={collapsible ? () => onPressGroupHeader(group, index) : null}
+            >
+                {group.headerRenderMode === "text" ? (
+                    <Text style={[style.header[group.headerTextRenderMode], { flex: 1 }]}>
+                        {group.headerText.value}
+                    </Text>
+                ) : (
+                    group.headerContent
+                )}
+                {icon !== "no" && collapsible && (
```

Comment:
I'm not a fan of using && as a shorthand for ternaries. It's easy to fall into a trap and get unexpected results. It can be a bit awkward to have a dangling `null` when you use a ternary of course, but it's semantically more accurate and doesn't rely on funny Javascript behavior.

Reply:
That makes sense, updated